decaying_enemies/decaying_enemies/Assets/Scriipts/decaying_enemy.cs

```diff
@@ -0,0 +1,36 @@
+using UnityEngine;
+public class decaying_enemy : MonoBehaviour
+{
+    [SerializeField] public float velocidad;
+    [SerializeField] public float velocidaddeReversa;
+    [SerializeField] public float distanciaDeljugador;
+    [SerializeField] public float rangodeVision;
+    [SerializeField] public float RangodeReversa;
+    [SerializeField] public float RangodeDisparo;
+    [SerializeField] public Transform player;
+    [SerializeField] public Rigidbody2D rb2D;
```

Comment:
The purpose of use [Seralizefield] is you don´t need to put the variables in public to see it in the inspector, so please change the access modifier to private and add to all a default value.